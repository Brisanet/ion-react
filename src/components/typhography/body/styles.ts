import stitches from '../../../stitches.config';
const { styled } = stitches;

export const Body = styled('p', {
  margin: 0,
  padding: 0,
  variants: {
    color: {
      primary: { color: '$neutral7' },
      secondary: { color: '$neutral6' },
      'dark-primary': { color: '$neutral3' },
      'dark-secondary': { color: '$neutral4' },
    },
    weight: {
      normal: { fontWeight: '400' },
      medium: { fontWeight: '600' },
      bold: { fontWeight: '700' },
    },
    style: { normal: { fontStyle: 'normal' }, italic: { fontStyle: 'italic' } },
    size: {
      small: { fontSize: '14px', lineHeight: '20px' },
      normal: { fontSize: '16px', lineHeight: '24px' },
    },
  },
});
