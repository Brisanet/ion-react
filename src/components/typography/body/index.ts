export * from './body';
export * from './styles';
