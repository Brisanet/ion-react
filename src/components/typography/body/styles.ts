import { css, styled } from 'styled-components';
import {
  IonBodyInterface,
  IonTypographyColorScheme,
  IonTypographyStyle,
  IonTypographyWeight,
} from '../typography.types';
import { defaultColor } from '../utils-typography';

type BodyStyleProps = {
  $color: IonTypographyColorScheme;
  $size: IonBodyInterface['size'];
  $style: IonTypographyStyle;
  $weight: IonTypographyWeight;
};

export const weights = {
  normal: 400,
  medium: 600,
  bold: 700,
};

export const sizes = {
  small: 14,
  normal: 16,
};

export const fontStyle = {
  normal: 'normal',
  italic: 'italic',
};

export const Body = styled.p<BodyStyleProps>`
  ${({ $color, $size, $style, $weight }) => css`
    margin: 0;
    padding: 0;
    color: ${defaultColor('body')[$color].color};
    ${$size && sizes[$size]};
    font-weight: ${weights[$weight]};
    font-style: ${fontStyle[$style]};
  `};
`;
