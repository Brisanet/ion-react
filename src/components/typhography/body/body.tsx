import React from 'react';
import { IonBodyInterface } from '../typhography.types';
import * as S from './styles';

export const IonBody: React.FC<IonBodyInterface> = ({
  label,
  colorScheme = 'primary',
  size = 'normal',
  style = 'italic',
  weight = 'normal',
}: IonBodyInterface) => {
  const props = { color: colorScheme, size, style, weight };
  return (
    <S.Body {...props} data-testid="ion-body">
      {label}
    </S.Body>
  );
};
