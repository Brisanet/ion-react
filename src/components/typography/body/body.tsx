import { IonBodyInterface } from '../typography.types';
import * as S from './styles';

export const IonBody = ({
  label,
  colorScheme = 'primary',
  size = 'normal',
  style = 'normal',
  weight = 'normal',
}: IonBodyInterface) => {
  const props = {
    $color: colorScheme,
    $size: size,
    $style: style,
    $weight: weight,
  };
  return (
    <S.Body {...props} data-testid='ion-body'>
      {label}
    </S.Body>
  );
};
