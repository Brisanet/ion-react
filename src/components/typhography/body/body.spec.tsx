import React from 'react';
import { render, screen } from '@testing-library/react';
import { IonBody, IonBodyProps } from './body';

const defaultProps: IonBodyProps = { label: 'Ion React' };
const sut = (props: IonBodyProps = defaultProps) => {
  return render(<IonBody {...props} />);
};

const bodyStyles: Array<IonBodyProps['style']> = ['italic', 'normal'];
const bodySize: Array<IonBodyProps['size']> = ['small', 'normal'];
const bodyWeight: Array<IonBodyProps['weight']> = ['normal', 'medium', 'bold'];
const bodyColor: Array<IonBodyProps['color']> = [
  'primary',
  'secondary',
  'dark-primary',
  'dark-secondary',
];

describe('<IonBody />', () => {
  it('should render the body with the correct label', () => {
    sut();
    expect(screen.getByTestId('ion-body')).toHaveTextContent(
      defaultProps.label
    );
  });
  it.each(bodySize)(
    'should render the body with the correct size %s',
    (size) => {
      const { container } = sut({ ...defaultProps, size });
      const element = container.firstChild as Element;
      expect(element.className).toContain(`size-${size}`);
      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each(bodyWeight)(
    'should render the body with the correct weight %s',
    (weight) => {
      const { container } = sut({ ...defaultProps, weight });
      const element = container.firstChild as Element;
      expect(element.className).toContain(`weight-${weight}`);
      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each(bodyColor)(
    'should render the body with the correct color %s',
    (color) => {
      const { container } = sut({ ...defaultProps, color });
      const element = container.firstChild as Element;
      expect(element.className).toContain(`color-${color}`);
      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each(bodyStyles)(
    'should render the body with the correct styles %s',
    (style) => {
      const { container } = sut({ ...defaultProps, style });
      const element = container.firstChild as Element;
      expect(element.className).toContain(`style-${style}`);
      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
