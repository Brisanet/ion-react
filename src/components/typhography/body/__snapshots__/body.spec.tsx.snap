// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IonBody /> should render the body with the correct color dark-primary 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-khXpfY-color-dark-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct color dark-secondary 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-kwrJpL-color-dark-secondary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct color primary 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct color secondary 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-jOtQRh-color-secondary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct size normal 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct size small 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cSNgox-size-small"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct styles italic 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct styles normal 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-ZzHSU-style-normal c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct weight bold 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-gBWVhb-weight-bold c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct weight medium 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-GgrlW-weight-medium c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;

exports[`<IonBody /> should render the body with the correct weight normal 1`] = `
<p
  class="c-fXXVzC c-fXXVzC-fTEIYA-color-primary c-fXXVzC-cdWBIM-weight-normal c-fXXVzC-bYNGXt-style-italic c-fXXVzC-cHYCsH-size-normal"
  data-testid="ion-body"
>
  Ion React
</p>
`;
